src/Promise/ErrorHandler.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Interop\Async\Promise;
+
+use Interop\Async\Promise;
+
+/**
+ * Global error handler for promises.
+ *
+ * Callbacks passed to `Promise::when()` should never throw, but they might. Such errors have to be passed to this
+ * global error handler to make them easily loggable. These can't be handled gracefully in any way, so we just enable
+ * logging with this handler and ignore them otherwise.
+ *
+ * If handler is set or that handler rethrows, it will fail hard by triggering an E_USER_ERROR leading to script
+ * abortion.
+ */
+final class ErrorHandler
+{
+    /** @var callable|null */
+    private static $callback = null;
+
+    private function __construct()
+    {
+        // disable construction, only static helper
+    }
+
+    /**
+     * Set a new handler that will be notified on uncaught errors during promise resolution callback invocations.
+     *
+     * This callback can attempt to log the error or exit the execution of the script if it sees need. It receives the
+     * exception as first and only parameter.
+     *
+     * As it's already a last chance handler, the script will be aborted using E_USER_ERROR if the handler throws. Thus
+     * it's suggested to always wrap the body of your callback in a generic `try` / `catch` block, if you want to avoid
+     * that.
+     *
+     * @param callable|null $onError Callback to invoke on errors or `null` to reset.
+     *
+     * @return void
+     */
+    public static function set(callable $onError = null)
+    {
+        self::$callback = $onError;
```

Comment:
Good point, we should do that there too. [I honestly thought we already were doing that.]